cpp/src/arrow/csv/converter_benchmark.cc

```diff
@@ -79,8 +79,7 @@ static void BenchmarkConversion(benchmark::State& state,  // NOLINT non-const re
   std::shared_ptr<Converter> converter = *Converter::Make(type, options);
 
   while (state.KeepRunning()) {
-    std::shared_ptr<Array> result;
-    ABORT_NOT_OK(converter->Convert(parser, 0 /* col_index */, &result));
+    ASSIGN_OR_ABORT(auto result, converter->Convert(parser, 0 /* col_index */));
```

Comment:
Why not simply `auto result = *converter->Convert(parser, 0 /* col_index */);`?

Reply:
It seemed better to be explicit than let someone hunt on a `ValueOrDie` termination

Reply:
Don't you get the same kind of message in both cases?

Reply:
I think so. I'll just revert it